packages/hash/frontend/src/blocks/page/ComponentView.tsx

```diff
@@ -42,13 +42,13 @@ const BLANK_PROPERTIES = {};
 
 const getChildEntity = (
   entity: DraftEntity | null | undefined,
-): DraftEntity<BlockEntity["properties"]["entity"]> | null => {
-  if (entity) {
+): DraftEntity<BlockEntity["dataEntity"]> | null => {
```

Comment:
I've asked about this before, but it seems like the backend team are happy with these kind of "named" shortcut resolvers for links. It would be good to agree a collective approach. https://github.com/hashintel/hash/pull/1060#discussion_r971044213

Reply:
We had previously agreed with this sentiment:
> I think we're probably better off just using links and having some functions clientside that construct whatever trees we want, to make the resolution of data less mysterious

([internal Slack link](https://hashintel.slack.com/archives/C03F7V6DU9M/p1661177030351259?thread_ts=1661176442.884289&cid=C03F7V6DU9M), no further context except agreement)

I am not firmly opposed to these helpers, but I think their existence encourages the abstraction to bleed further across the stack than it needs to. I _am_ firmly opposed (absent any good argument, which might come up) for it being something that's handled in the save process, because I think it's introducing special casing that is not required and makes the whole system harder to understand and maintain.